Include/internal/pycore_descrobject.h

```diff
@@ -0,0 +1,26 @@
+#ifndef Py_INTERNAL_DESCROBJECT_H
```

Comment:
Maybe "pycore_property.h" as it is just the property struct.

Reply:
I'm leaving it generic from now (and naming it after the file it came from) so that I can re-use it again in the future when I add more specialisations. (E.g. classmethod is also from the same file)

Reply:
Do properties and classmethods really belong in the same file? Sure, they are both descriptors, but so are Python functions.
The file layout is a bit of an historical artifact.

Reply:
If you don't want to change it for this PR, that's fine too.